pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -492,7 +492,18 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
                     "The flag `devices` must be an int with `accelerator='cpu'`,"
                     f" got `devices={self._devices_flag}` instead."
                 )
-
+        if not self.accelerator.is_available():
+            if isinstance(self.accelerator, GPUAccelerator):
+                hardware_str = "GPU"
```

Comment:
Won't be better to add hardware_str to the Accelerator directly?

Reply:
+1, I would expect to have a `hardware_name` just as we do for `strategy_name`.

Fine doing this in a follow-up if you prefer

Reply:
@tchaton @carmocca updated, is it similar to what you had in mind?